Data/DataModel/PuzzleStatePerTeam.cs

```diff
@@ -16,7 +16,45 @@ public class PuzzleStatePerTeam
 
         public virtual Puzzle Puzzle { get; set; }
         public virtual Team Team { get; set; }
-        public virtual State State { get; set; }
-        public DateTime StateChanged { get; set; }
+
+        /// <summary>
```

Comment:
@asyasky, what motivated you to have this stuff in a separate table?

Reply:
Originally PuzzleStatePerTeam was a join table with a timestamp. I was keeping that pattern when I re-created it. Let me thing about what makes the most sense in this case and then we can have a brief design discussion.

Reply:
I feel like the PuzzleStatePerTeam and State tables had a 1:1 mapping, and if so I don't see the advantage in having separate tables.

Reply:
Just so we're clear: I am blocked on this issue. What I've done makes sense to me but I'm not a DB person...